mapslang/Makefile

```diff
@@ -1,27 +1,58 @@
-# Project: adikted
-
-#To create a debug executable, run with the command:
-# make DEBUG
-
-#ON_LINUX = 1
-
-ifneq (,$(ON_LINUX))
+# Project: mapslang
+# Makefile created by KeeperFX/dkfans team
+
+# Instructions:
+# To create the Windows executable run:
+# make all
+# To create the linux executable run:
+# make all BUILD_FOR_LINUX=1
+# To create a debug executable, run with the command:
+# make DEBUG (THIS IS NOT IMPLEMENTED YET)
+# To clean up the built files (objects, library, and resource file) run:
+# make clean
+
+# Tested and working with:
+# - Windows : MSYS2 (MinGW 32-bit) - make
+# - Windows : MSYS2 (MinGW 32-bit) - mingw32-make
+# - Windows : MinGW (MSYS) - make
+
+# Executable files extension on host environment
+ifneq (,$(findstring Windows,$(OS)))
+  CROSS_EXEEXT = .exe
+else
+  CROSS_EXEEXT =
   CROSS_COMPILE = i686-w64-mingw32-
-  SLANG_HOST=mingw32
+  SLANG_HOST = mingw32
 endif
-
-SLANG_URL = https://www.jedsoft.org/snapshots/slang-pre2.3.3-24.tar.gz
-SLANG_NAME=slang-pre2.3.3-24
-
-CC                = $(CROSS_COMPILE)gcc
-CPP                = $(CROSS_COMPILE)g++
-CFLAGS                          = -c -O -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1
-CFLAGS-DEBUG                    = -c -g -I. -I../lib/$(SLANG_NAME)/src -DSLANG_DLL=1
-LINK                = $(CROSS_COMPILE)gcc
-WINDRES                         = $(CROSS_COMPILE)windres
-LFLAGS                          = 
-LFLAGS-DEBUG                    = 
-LIBS                = -L. -L../libadikted -L../lib/$(SLANG_NAME)/src/gw32objs -lslang -ladikted
+# Executable files extension on target environment (
+#BUILD_FOR_LINUX = 1
+ifneq (,$(BUILD_FOR_LINUX))
+  EXEEXT =
+  OTHEREXEEXT = .exe
+  LIBEXT = .so
+  OTHERLIBEXT = .dll
+  SLANGPLATFORM =
+else
+  EXEEXT = .exe
+  OTHEREXEEXT =
+  LIBEXT = .dll
+  OTHERLIBEXT = .so
+  SLANGPLATFORM = WIN32 MINGW32 DLL
+endif
+# Names of utility commands
+SLANG_URL          = https://www.jedsoft.org/snapshots/slang-pre2.3.3-24.tar.gz
+SLANG_NAME         = slang-pre2.3.3-24
+CPP                = $(CROSS_COMPILE)g++$(CROSS_EXEEXT) -std=c18
+CC                 = $(CROSS_COMPILE)gcc$(CROSS_EXEEXT) -std=c18 -Wall -Wextra -Wno-pedantic -Wno-conversion -Wno-traditional-conversion -Wno-sign-compare
```

Comment:
Do these compilation flags work? Specifically the `-std` flag I remember trying out bumping to `c18`, only to have it blow up in my face, granted this was for libadikted and not mapslang. I'm not sure if it would be better to keep the `-std` flag in sync between libadikted and mapslang, or let mapslang use the newer `c18` standard and then down the line have libadikted also use the `c18` standard instead of the older `c89` standard.

Though keep the warning flags, these are always good to have enabled I feel.

Reply:
It does for me, I intend to have both as c89, but mapslang won't compile with c89 at the moment. I need to at least change the c++ comments to c comments. And maybe some other build errors. Then I will set both to c89